### ISSUE TITLE
refactor: remove ExportComputationGraph and migrate all layers to tape-based autodiff

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1059]
## Summary

Remove all `ExportComputationGraph` overrides from every layer and delete the method from `LayerBase<T>`. With the GradientTape (#1057) and named ports (#1058) in place, Forward itself records to the tape — the separate graph export path is dead code.

**Depends on:** #1057 (GradientTape rewrite), #1058 (Named ports)

## Problem

`ExportComputationGraph` is a parallel implementation of Forward that:
- Duplicates logic and diverges from Forward (proven: DiffusionResBlock dropped time conditioning)
- Is unimplemented in 7+ layers (VAEEncoder, VAEDecoder, VAEResBlock, NoisePredictorBase, DeepANT ConvLayerTensor, etc.)
- Forces SPSA fallback whenever any layer in the chain throws NotSupportedException
- Adds maintenance burden — every Forward change requires a matching graph change

With tape-based autodiff, Forward IS the graph. ExportComputationGraph has no purpose.

### Layers with ExportComputationGraph overrides to remove

Search pattern: `override.*ExportComputationGraph` across src/

**Layers that throw NotSupportedException (stubs):**
- [ ] VAEEncoder.ExportComputationGraph
- [ ] VAEDecoder.ExportComputationGraph
- [ ] VAEResBlock.ExportComputationGraph
- [ ] NoisePredictorBase.ExportComputationGraph
- [ ] DeepANT ConvLayerTensor.ExportComputationGraph

**Layers with actual (but potentially stale) graph implementations:**
- [ ] DenseLayer.ExportComputationGraph
- [ ] DiffusionResBlock.ExportComputationGraph
- [ ] Conv2DLayer.ExportComputationGraph
- [ ] BatchNormalizationLayer.ExportComputationGraph
- [ ] ActivationLayer.ExportComputationGraph
- [ ] All other layers with graph overrides (full audit needed)

### Base class changes
- [ ] Remove `ExportComputationGraph` from `LayerBase<T>` (or mark `[Obsolete]` if phased)
- [ ] Remove `SupportsJitCompilation` property (replaced by tape availability)
- [ ] Remove `ILayer<T>.ExportComputationGraph` from interface

### Callers to update
- [ ] `DiffusionModelBase.ComputeGradients` — replace autodiff block with `GradientTape` around `PredictNoise` call (Forward records to tape automatically)
- [ ] `VAEModelBase.ComputeGradients` — same pattern
- [ ] `NeuralNetworkBase` JIT compilation path — if any
- [ ] Any other callers of `ExportComputationGraph`

### Example: DiffusionModelBase after migration

```csharp
public virtual Vector<T> ComputeGradients(Tensor<T> input, Tensor<T> target, ILossFunction<T>? lossFunction)
{
    var parameters = GetParameters();

    using var tape = new GradientTape<T>();

    // Forward records to tape automatically — no ExportComputationGraph needed
    var noisySample = AddNoise(input, noise, timestep);
    var predicted = PredictNoise(
        new Dictionary<string, Tensor<T>>
        {
            ["noisy_sample"] = noisySample,
            ["timestep"] = Tensor.Scalar(timestep),
            ["conditioning"] = conditioning
        });

    // Compute loss
    var diff = Engine.TensorSubtract(predicted, noise);  // Recorded to tape
    var loss = Engine.TensorMean(Engine.TensorMultiply(diff, diff));  // Recorded

    // Backward — exact gradients through the full PredictNoise path
    var grads = tape.Gradient(loss);

    // Extract parameter gradients
    return ExtractParameterGradients(grads);
}
```

### Step 1: Audit all ExportComputationGraph overrides
- [ ] Full grep across codebase for `ExportComputationGraph`
- [ ] Categorize each as: stub (throws), stale (diverged from Forward), or correct
- [ ] Document which layers have graph code worth preserving as reference for backward functions

### Step 2: Update callers
- [ ] `DiffusionModelBase.ComputeGradients` — use tape around Forward, remove SPSA fallback for layers with tape support
- [ ] `VAEModelBase.ComputeGradients` — same
- [ ] Keep SPSA as fallback ONLY for non-differentiable ops (sampling, discrete operations)

### Step 3: Remove ExportComputationGraph
- [ ] Remove from `ILayer<T>` interface
- [ ] Remove abstract/virtual declaration from `LayerBase<T>`
- [ ] Remove `SupportsJitCompilation` property
- [ ] Remove all overrides across all layers
- [ ] Remove unused `Autodiff.ComputationNode<T>` types if no longer needed

### Step 4: Clean up SPSA
- [ ] SPSA stays as fallback for non-differentiable operations only
- [ ] Add `SupportsTapeAutodiff` property (true for all standard layers, false for stochastic layers)
- [ ] Log when SPSA is used so developers know which layer blocked autodiff

## Acceptance Criteria
- No `ExportComputationGraph` overrides remain in any layer
- DiffusionModelBase computes gradients via tape through the full PredictNoise path (including timestep)
- VAEModelBase computes gradients via tape through encoder + decoder
- All existing gradient-dependent tests pass
- SPSA is only used as explicit fallback for documented non-differentiable operations

## Risk Mitigation
- Run gradient checking (finite-difference vs tape) for each migrated component
- Compare SPSA gradients vs tape gradients to verify they agree in direction (not magnitude — SPSA is noisy)
- Keep SPSA code available but gated behind a flag for A/B comparison during validation

## References
- PyTorch autograd internals: https://pytorch.org/blog/overview-of-pytorch-autograd-engine/
- "Automatic Differentiation in Machine Learning" (Baydin et al. 2018): https://arxiv.org/abs/1502.05767
